### DISCUSSION TITLE
Chat buffer is being locked after AI response

### DISCUSSION THREAD

**Discussion by rpbaptist** [olimorris/codecompanion.nvim#361] [Q&A] [answered]
I did the minimal config and I can't reproduce the issue. The fact is that I'm using LazyVim and apparently something else is preventing the buffer from unlocking after the response has been delivered.

The issue seems to be that control is not handed back to the "Me" after a response is generated.

The buffer is locked and I don't see the `## Me` header after `## CodeCompanion` has returns it's output.

I realize this is specific to my config, but I would like some pointers on how to get to the cause of this.

The log file isn't providing additional info here.

**Comment by olimorris** [comment]
Sorry to hear this. Unfortunately with the amount of plugins we all have installed, it can be like trying to find a needle in a haystack.

I'd advise disabling any markdown specific plugins as a first action. If that doesn't work, I'd advise disabling all of them and adding them back as you go. I've had to do this with a few plugins and it is nothing but painful.

**Reply by rpbaptist** [reply, marked as answer]
Thanks for your response. Well, I decided something else and set up lazy to just get latest git commits instead of semversioning. It updated a whole lot of packages and the issue is resolved.

**Comment by bashtoni** [comment]
For anyone else experiencing this issue, after much frustration I finally tracked it down.

The issue is with plenary.nvim - I believe specifically https://github.com/nvim-lua/plenary.nvim/pull/575

As this is a direct dependency of codecompanion, *everyone* who is using Lazy and has `version = "*"` set will be affected.

To resolve, I changed the plenary line in `dependencies` to:

```lua
{ "nvim-lua/plenary.nvim", version = "2d9b061" },
```

@olimorris - maybe worth updating the README to mention this?

**Reply by olimorris** [reply]
I'll have to dig into this. But one thing I don't like is pinning to a specific version of such a widely used library.

Can anyone provide a `minimal.lua` file that recreates this? I've never had this happen in the last 9 months of using the plugin with Plenary.

**Reply by bashtoni** [reply]
I just raised https://github.com/olimorris/codecompanion.nvim/issues/377 with this.
